### DISCUSSION TITLE
Write a Monarch JSON parser

### DISCUSSION THREAD

**Discussion by troy351** [microsoft/monaco-editor#3753] [Q&A] [answered]
It seems the `JSON` language's token was integrated into `jsonMode.js` and can not be found in [monaco-languages](https://github.com/microsoft/monaco-languages) like other languages does.
Is it possible?

**Comment by troy351** [comment]
This makes overwrite tokenize impossible (when integrated with `vscode-textmate`)

**Comment by alexdima** [comment]
Yes, JSON is the only language we ship that has its tokenization defined via a "manual" tokenizer and not via a Monarch tokenizer. Since it is such an easy language, it might make sense, for consistency purposes, to also write a Monarch JSON parser.

**Comment by troy351** [comment]
Any workaround how to replace current `built-in` json tokenizer?

**Comment by alexdima** [comment, marked as answer]
Yes, it looks like you can use `monaco.languages.json.jsonDefaults.setModeConfiguration({ tokens: false });`

**Comment by troy351** [comment]
It works, Thanks!

**Comment by kartik-gupta-ij** [comment]
@alexdima Is Monarch JSON parser available?
